test/utility.h

```diff
@@ -118,10 +128,37 @@ class TestContext : public ContextBase {
 
 class TestWasm : public WasmBase {
 public:
-  TestWasm(std::unique_ptr<WasmVm> wasm_vm, std::unordered_map<std::string, std::string> envs = {})
-      : WasmBase(std::move(wasm_vm), "", "", "", std::move(envs), {}) {}
+  TestWasm(std::unique_ptr<WasmVm> wasm_vm, std::string_view vm_id,
+           std::string_view vm_configuration, std::string_view vm_key,
+           std::unordered_map<std::string, std::string> envs,
+           AllowedCapabilitiesMap allowed_capabilities)
+      : WasmBase(std::move(wasm_vm), vm_id, vm_configuration, vm_key, std::move(envs),
+                 std::move(allowed_capabilities)) {}
+
+  TestWasm(std::unique_ptr<WasmVm> wasm_vm, std::unordered_map<std::string, std::string> envs)
+      : TestWasm(std::move(wasm_vm), "", "", "", std::move(envs), {}) {}
+
+  TestWasm(std::unique_ptr<WasmVm> wasm_vm) : TestWasm(std::move(wasm_vm), {}) {}
```

Comment:
Is the 1-arg unique_ptr<WasmVm> constructor used anywhere?

(optional) 4 constructors seems like a lot. I wonder if you could get away with just 2, i.e. the 6-arg unique_ptr<WasmVm> one and the 3-arg shared_ptr<WasmHandleBase> one.

Reply:
1-arg unique_ptr is used in the existing codes.
(e.g., https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/2cf68d6c00b4141ecf2afb5e5d51ebd21a27ed43/test/exports_test.cc#L77)

BTW, I reduced them from 5 to 3. 
2-arg is also possible, but 3-arg seems to be slightly cleaner when using TestWasm class in test codes.